nuxeo-features/nuxeo-automation/nuxeo-automation-features/src/test/java/org/nuxeo/ecm/automation/core/operations/services/bulk/TestSetPropertiesActionOptimizations.java

```diff
@@ -0,0 +1,183 @@
+
+        // make sure all previous logs have been bulked and processed
+        logs.await(60, TimeUnit.SECONDS);
+
+        int initialLogEntries = logs.getLogEntriesFor(note.getId(), session.getRepositoryName()).size();
+
+        bulkService.submit(createBuilder().build());
+        assertTrue("Bulk action didn't finish", bulkService.await(Duration.ofSeconds(60)));
+        assertTrue("Logs didn't finish", logs.await(60, TimeUnit.SECONDS));
+        // documentCheckin and documentModified
+        assertEquals(initialLogEntries + 2, logs.getLogEntriesFor(note.getId(), session.getRepositoryName()).size());
+
+        bulkService.submit(createBuilder().param(PARAM_DISABLE_AUDIT, TRUE).build());
+        assertTrue("Bulk action didn't finish", bulkService.await(Duration.ofSeconds(60)));
+        assertTrue("Logs didn't finish", logs.await(60, TimeUnit.SECONDS));
+        // only documentCheckin
+        assertEquals(initialLogEntries + 3, logs.getLogEntriesFor(note.getId(), session.getRepositoryName()).size());
+
+        bulkService.submit(createBuilder().param(PARAM_VERSIONING_OPTION, NONE.toString()).build());
+        assertTrue("Bulk action didn't finish", bulkService.await(Duration.ofSeconds(60)));
+        assertTrue("Logs didn't finish", logs.await(60, TimeUnit.SECONDS));
+        // only documentModified
+        assertEquals(initialLogEntries + 4, logs.getLogEntriesFor(note.getId(), session.getRepositoryName()).size());
+
+        bulkService.submit(createBuilder().param(PARAM_DISABLE_AUDIT, TRUE)
+                                          .param(PARAM_VERSIONING_OPTION, NONE.toString())
+                                          .build());
+        assertTrue("Bulk action didn't finish", bulkService.await(Duration.ofSeconds(60)));
+        assertTrue("Logs didn't finish", logs.await(60, TimeUnit.SECONDS));
+        // none
+        assertEquals(initialLogEntries + 4, logs.getLogEntriesFor(note.getId(), session.getRepositoryName()).size());
+    }
+
+    @Test
+    public void testDisableVersioning() throws InterruptedException {
+        assertEquals(1, session.getVersions(note.getRef()).size());
+        // should not create a new version
+        bulkService.submit(createBuilder().param(PARAM_VERSIONING_OPTION, NONE.toString()).build());
+        assertTrue("Bulk action didn't finish", bulkService.await(Duration.ofSeconds(60)));
+        transactions.nextTransaction();
+        assertEquals(1, session.getVersions(note.getRef()).size());
+        // should create a new version
+        bulkService.submit(createBuilder().build());
+        assertTrue("Bulk action didn't finish", bulkService.await(Duration.ofSeconds(60)));
+        transactions.nextTransaction();
+        assertEquals(2, session.getVersions(note.getRef()).size());
+    }
+
+    @Test
+    public void testParamValues() throws InterruptedException {
+        // audit param
+        bulkService.submit(createBuilder().param(PARAM_DISABLE_AUDIT, null).build());
```

Comment:
Furthermore, submitting so many commands to bulk service and then waiting for them to test that valid/invalid parameters doesn't crash the computation is too expensive. This will result to more unnecessary load in CI.

Reply:
the expensive part is initializing the test feature, creating CQ files is now cheap since NXP-25626,
submitting command & processing should be fast and light, maybe the await on all bulk command is slow because relying on long sleep interval intentionally.
that being said it is better to unit test param checks without an overhead

Reply:
Ok for load to CI.
Agree on overhead, could be better to unit test param check directly indeed.